### PR TITLE
Mercadopagoapi

### PR DESCRIPTION
**Pull request by alejandromartinezcs**
Hello, this pull request has the necessary route to be able to checkout with mercadopago, doubts arise: 

1.- It is a route because it is an API, but it is also a new library, does it go in routes or does it go in lib, following the clean architecture.
2.- Which is the safe way to pass the .env components, which are the best practices?
3.- Is it necessary to create a model to also store the information of the purchases of the users, not the cards, only the quantity of items purchased and the amounts, do I link both?

### CODE REVIEW

File: src/routes/paymentsCheckout.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const express = require("express");
+const mercadopago = require("mercadopago");
+const router = express.Router();
+require("dotenv").config();
+
+
+router.post("/payment_mercadopago", (req, res) => {
+    mercadopago.configurations.setAccessToken(process.env.ACCESS_TOKEN_MP);
```
Comment (emiliodeleonhdez) [inline src/routes/paymentsCheckout.js]: env variables can be declared in the config.js file, then you will need to import the config module to this file

File: src/routes/paymentsCheckout.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const express = require("express");
+const mercadopago = require("mercadopago");
+const router = express.Router();
+require("dotenv").config();
+
+
+router.post("/payment_mercadopago", (req, res) => {
+    mercadopago.configurations.setAccessToken(process.env.ACCESS_TOKEN_MP);
+    const payment_data = {
```
Comment (emiliodeleonhdez) [inline src/routes/paymentsCheckout.js]: Let's split this function a usecase. There we can just pass the request body in a single const

File: src/routes/paymentsCheckout.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+router.post("/payment_mercadopago", (req, res) => {
+    mercadopago.configurations.setAccessToken(process.env.ACCESS_TOKEN_MP);
+    const payment_data = {
+      transaction_amount: req.body.transaction_amount,
+        token: req.body.token,
+        description: req.body.description,  
+        installments: Number(req.body.installments),
+        payment_method_id: req.body.paymentMethodId,
+        issuer_id: req.body.issuer,
+        // notification_url: "test"
```
Comment (emiliodeleonhdez) [inline src/routes/paymentsCheckout.js]: no comment lines

File: src/routes/paymentsCheckout.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                type: req.body.payer.docType,
+                number: req.body.payer.docNumber,
+            },
+        },
+    };
+
+    mercadopago.payment
+        .save(payment_data)
+        .then((response) => {
+            return res.status(response.status).json({
```
Comment (emiliodeleonhdez) [inline src/routes/paymentsCheckout.js]: we can use axios to manage the promises: https://www.npmjs.com/package/axios

**Review by emiliodeleonhdez — CHANGES_REQUESTED** [review]
Resolve comments when possible

File: src/routes/paymentsCheckout.js (hunk 4 above)
Comment (alejandromartinezcs) [reply src/routes/paymentsCheckout.js]: Hi @emiliodeleonhdez , if we use axios, should it look similar to this?:
`const payment = axios.post (url, body)
.then((response) => {" do something if response is 200"} else {"payment not successful "})
.catch((error) => { "do something whit error"});`

File: src/routes/paymentsCheckout.js (hunk 3 above)
Comment (alejandromartinezcs) [reply src/routes/paymentsCheckout.js]: Done, comment remove.

File: src/routes/paymentsCheckout.js (hunk 2 above)
Comment (alejandromartinezcs) [reply src/routes/paymentsCheckout.js]: Hi Emi, I have a doubts about this. 

Regarding the construction of the array of products, should we obtain the data from the shopping cart, right? With that said, would a shopping cart model be created? And here, would an await function be waiting to obtain the shopping carts? 

I suppose the array would be filled when the user clicks on "pay", the data would be pulled from there, the array would be filled, and the body would be passed. 

If that's the case, would this be the correct logic?

File: src/usecases/payment/index.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  }));
+
+  const stagingUser = {
+    first_name: "TEST_USER_1310746633",
+    last_name: "Test",
+    address: {},
+  };
+
+  const backurls = {
+    success: `${config.url.front}/success`,
```
Comment (emiliodeleonhdez) [inline src/usecases/payment/index.js:30]: Make sure this is pointing to fronts deploy

**Review by emiliodeleonhdez — APPROVED** [review]
Left one comment

**Pull request merged**